### PR TITLE
Add Linux native build

### PR DESCRIPTION
**Pull request by alvinhochun**
Depends on https://github.com/microsoft/cppwinrt/pull/1238 and https://github.com/microsoft/winmd/pull/28

This adds support for building a native Linux executable of cppwinrt. It works mostly the same as on Windows, except without the code that searches for Windows SDKs and local WinMetadata.

### CODE REVIEW

**Comment by alvinhochun** [comment]
I did test the native Linux build with a copy of `Windows.winmd` from Windows SDK 19041 and it outputs the exact same headers as a native Windows build. But I can't add a CI check for this because there is no way to get the `Windows.winmd` file for use on Linux. Is there any way to get this winmd file without having to install the full Windows SDK?

**Comment by kennykerr** [comment]
Almost forgot - there is a nuget package with just the winmd files:

https://www.nuget.org/packages/Microsoft.Windows.SDK.Contracts/

**Comment by kennykerr** [comment]
Right, everything in the `windows-rs` repo is dual licensed so that would include the winmd files.

**Comment by kennykerr** [comment]
I sent you an invite to give you permission to re-run workflows as needed.

**Comment by alvinhochun** [comment]
> I sent you an invite to give you permission to re-run workflows as needed.

Thanks! Looking at the latest failure, I think I should try using [actions/cache@v3](https://github.com/actions/cache) action to avoid downloading the same dependencies repeatedly. Has this action been allowed in the repo settings?

**Comment by kennykerr** [comment]
Yes, that's what I use on windows-rs - I've enabled it here as well.

**Comment by alvinhochun** [comment]
> Something seems to be wrong with the test validation. It doesn't seem to break the build anymore.
> 
> https://github.com/microsoft/cppwinrt/actions/runs/3628603087/jobs/6120073812#step:7:44

This test is marked for Clang "shouldfail" because libc++ does not yet support `source_location`, so this is expected.

File: cppwinrt/cmd_reader.h [hunk 1]
```diff
@@ -3,6 +3,7 @@
 #include <cassert>
 #include <array>
 #include <limits>
+#include <climits>
```
[the remark this answers is not in the document]
Comment (alvinhochun) [reply cppwinrt/cmd_reader.h:6]: Line 316 has `static constexpr uint32_t no_max = UINT_MAX;`

**Pull request merged**